pkgs/by-name/taldir/package.nix

```diff
@@ -4,21 +4,45 @@
   recutils,
   buildGoModule,
 }:
-buildGoModule {
+buildGoModule (finalAttrs:{
   pname = "taldir";
-  version = "0-unstable-2024-02-18";
+  version = "1.0.3";
 
   src = fetchgit {
     url = "https://git.taler.net/taldir.git";
-    rev = "9c5230f64d16d46c000c6d4f5842170c51b697ce";
-    hash = "sha256-Ri0R7kxP/FitBRKtrM48Cbks63mqpXsRc3r98M0sMus=";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-axVQ687cGvxGEKECh3HmbTAFwI/YaCtPvtRU5bWsYYI=";
   };
 
   postPatch = ''
     cp ${./go.sum} go.sum
   '';
+/*  
+  Steps to fix outdated go.sum in Nix builds:
 
-  vendorHash = "sha256-yN8CiRK7cS4bHndOcu+/HI50PDOG+5x/t2kxlIt+5Mk=";
+  1. Set up a Go environment using Nix:
+      nix-shell -p go
+
+  2. Clone the Taldir repository and navigate to it:
+      git clone https://git.taler.net/taldir.git
+      cd taldir
+      make && make install
+
+  3. Build the project from source to regenerate go.sum:
+      go mod tidy
+
+  4. Verify that go.sum is created and updated:
+      cat go.sum
+
+  5. Copy the new go.sum file into the Nix package directory.
+
+  6. Retry the Nix build to get the correct vendorHash:
+      nix build .#packages.x86_64-linux.taldir
+
+  This ensures that Nix can properly resolve Go dependencies and avoids hash mismatch errors.
+*/
+
+  vendorHash = "sha256-V/UzUS3eMKnEhgaipsHoodAlZxuEkZM/ALliU1TNuYg=";
```

Comment:
Could you move `vendorHash` to be positioned right after `src`? It's nice to keep them close so it's easier for you to update them.